zebra-rpc/src/methods/get_block_template_rpcs/types/get_block_template.rs

```diff
@@ -9,6 +9,40 @@ use crate::methods::{
     GetBlockHash,
 };
 
+#[derive(Debug, serde::Deserialize)]
+#[serde(rename_all = "lowercase")]
+pub enum GetBlockTemplateRequestMode {
+    Template,
+    Proposal,
+}
+
+impl Default for GetBlockTemplateRequestMode {
+    fn default() -> Self {
+        Self::Template
+    }
+}
+
+#[derive(Debug, serde::Deserialize)]
+#[serde(rename_all = "lowercase")]
+pub enum GetBlockTemplateCapability {
+    LongPoll,
+    CoinbaseTxn,
+    CoinbaseValue,
+    Proposal,
+    ServerList,
+    WorkId,
+}
+
+/// Optional `jsonparametersobject` argument
+#[derive(Debug, serde::Deserialize)]
+pub struct JsonParameters {
+    #[serde(default)]
+    pub mode: GetBlockTemplateRequestMode,
```

Comment:
I think mode is optional according to the spec?
https://en.bitcoin.it/wiki/BIP_0022#Block_Template_Request

I understand that serde substitutes the default, but that also needs to be documented.

Reply:
Yeah, updated docs to say "all other fields are optional" in [`575deba`](https://github.com/ZcashFoundation/zebra/pull/5772/commits/575debad65065ec49e332709dad682847f3c9492).